src/fragments/ReactionRenderer.tsx

```diff
@@ -0,0 +1,68 @@
+import {  Message, ReactionSummary } from 'chatkitty';
+import { ChatAppContext } from 'providers/ChatAppProvider';
+import React, { useContext } from 'react';
+import { StyledBox } from 'react-chat-ui-kit';
+
+
+
+interface EmojiProps {
+    message: Message
+}
+
+const ReactionRenderer: React.FC<EmojiProps> = ({
+    message
+}: EmojiProps) => {
+
+    const {currentUser, messageUnReactor, messageReactor} = useContext(ChatAppContext);
+
+    const emojiClickListener = (reaction: ReactionSummary) => {
+        let notIn = true;
+        let reacted = false;
+        if(message.reactions && currentUser){
+            for(let i = 0; i< message.reactions?.length; i++){
```

Comment:
Optional: This can be done with less code using `find`. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find